### DISCUSSION TITLE
I can't connect JDDevice

### DISCUSSION THREAD

**Discussion by ruyi-tsai** [microsoft/jacdac#888] [DDK]
I　try using python connect jacdac device 
show this message : 
devtools server connected
JD: starting bus, self=<JDDevice RV33>
JD: reattaching services to <JDDevice RV33>; 0/0 to attach
JD: clear attach cache
JD: reattaching services to <JDDevice RV33>; 0/0 to attach
I already download Jacdac STM32G0xx but I don't know how to connect this from websocket 
I connect USB on "Bus 001 Device 004: ID 0483:374b STMicroelectronics ST-LINK/V2.1" on desktop ubuntu

**Comment by jamesadevine** [comment]
Hi, 

You cannot connect directly to the Jacdac bus from a PC. You need a device that has USB to pass Jacdac packets over USB. This can be a micro:bit V2, an ESP, or RP2040.

**Reply by jamesadevine** [reply]
The device will have to be programmed from MakeCode which includes the Jacdac USB interface. Serial might work too with the ESP32.

**Comment by pelikhan** [comment]
On pi, you will be able to use the JacHAT bridge.
________________________________

**Comment by ruyi-tsai** [comment]
microsoft/jacdac-stm32x0
so stm32 not support jacdac interface ? 
I already download jacdac bootloader on my stm32 controller
and can send jacdac frame package each  500 ms (I thinck it control package)

if i want to use stm32 controller how can i do connect to webserve?

**Reply by pelikhan** [reply]
You can connect your device to a device that supports USB/Serial connection, like the micro:bit V2 + jacdaptor

**Comment by pelikhan** [comment]
@ruyi-tsai are you still blocked on this?

**Comment by ruyi-tsai** [comment]
Thanks  , I change my job so transfer to another's.

取得 Android 版 Outlook<https://aka.ms/AAb9ysg>
________________________________
